src/Microsoft.DotNet.ImageBuilder/src/Commands/BuildCommand.cs

```diff
@@ -137,12 +145,51 @@ private void PublishImageInfo()
 
                 platform.Digest = matchingBuiltPlatform.Digest;
                 platform.Created = matchingBuiltPlatform.Created;
+                platform.Components = matchingBuiltPlatform.Components;
             }
 
             string imageInfoString = JsonHelper.SerializeObject(_imageArtifactDetails);
             File.WriteAllText(Options.ImageInfoOutputPath, imageInfoString);
         }
 
+        private void SetComponents(PlatformData platform)
+        {
+            if (platform.Components.Any() ||
+                string.IsNullOrEmpty(Options.GetInstalledPackagesScriptPath) ||
+                platform.PlatformInfo is null ||
+                platform.PlatformInfo.Model.OS == OS.Windows)
+            {
+                return;
+            }
+
+            // Use the platform's overriden script path if it's defined in the manifest; otherwise fall back to the default script.
+            string? getInstalledPackagesScriptPath = platform.PlatformInfo.Model.PackageQueryOverrides?.GetInstalledPackagesPath;
+            if (getInstalledPackagesScriptPath is null)
+            {
+                getInstalledPackagesScriptPath = Options.GetInstalledPackagesScriptPath;
+            }
+            else
+            {
+                getInstalledPackagesScriptPath = Path.Combine(Manifest.Directory, getInstalledPackagesScriptPath);
+            }
+
+            string imageTag = platform.PlatformInfo.Tags.First().FullyQualifiedName;
+
+            string args = $"{getInstalledPackagesScriptPath} {imageTag} {platform.PlatformInfo.DockerfilePath}";
+            string? output = _processService.Execute("/bin/sh", args, Options.IsDryRun);
+            if (output is not null)
+            {
+                platform.Components = output
+                    .Split(Environment.NewLine)
+                    .Select(val =>
+                    {
+                        string[] pkgVersion = val.Split(new char[] { ',', '=' });
```

Comment:
I think renaming this to something more abstract helps the readability especially when indexing into the various parts below.
```suggestion
                        string[] pkgInfo = val.Split(new char[] { ',', '=' });
```